TEditXna/Terraria/World.FileV2.cs

```diff
@@ -21,8 +21,10 @@ private static void SaveV2(World world, BinaryWriter bw)
         {
             world.Validate();
 
-            // initialize tileframeimportance array if empty
-            if (TileFrameImportant == null || TileFrameImportant.Length < TileCount)
+            if (TileFrameImportant.Length < TileCount && TileFrameImportant.Length > 0)

```

Comment:
If framed tiles change between versions there could be issues here. I don't know if this is something that can be fixed here without keeping track of framed tiles for every version.